src/lib/y2packager/release_notes_fetchers/url.rb

```diff
@@ -0,0 +1,360 @@
+          proxy = "--proxy #{Yast::Proxy.http}"
+          proxy << " --proxy-user '#{user_pass}'" unless user_pass.empty?
+        end
+
+        @curl_proxy_args = proxy
+      end
+
+      # Release notes index for the given product
+      #
+      # @return [Array<String>] Index containing the list of release notes files
+      # @see #download_release_notes_index
+      def release_notes_index
+        return @release_notes_index if @release_notes_index
+        @release_notes_index = download_release_notes_index(relnotes_url_base) || []
+      end
+
+      # Determine whether the release notes index contains an entry for the given
+      # language and format
+      #
+      # @return [Boolean]
+      def indexed_release_notes_for?(user_lang, format)
+        release_notes_index.include?(release_notes_file(user_lang, format))
+      end
+
+      # Download of index of release notes for a specific product
+      #
+      # @param url_base URL pointing to directory with the index
+      # @return [Array<String>,nil] filenames, nil if not found
+      def download_release_notes_index(url_base)
+        url_index = url_base + "/directory.yast"
+        log.info("Index with available files: #{url_index}")
+        tmpfile = Tempfile.new("directory.yast-")
+
+        # download the index with much shorter time-out
+        ok = curl_download(url_index, tmpfile.path, max_time: 30)
+
+        if ok
+          log.info("Release notes index downloaded successfully")
+          index_file = File.read(tmpfile.path)
+          if index_file.nil? || index_file.empty?
+            log.info("Release notes index empty, not filtering further downloads")
+            nil
+          else
+            rn_filter = index_file.split("\n")
+            log.info("Index of RN files at the server: #{rn_filter}")
+            rn_filter
+          end
+        elsif ok.nil?
+          nil
+        else
+          log.info "Downloading index failed, trying all files according to selected language"
+          nil
+        end
+      ensure
+        tmpfile.close
+        tmpfile.unlink
+      end
+
+      # Download *url* to *filename*
+      #
+      # May disable release notes downloading by calling .disable!.
+      #
+      # @return [Boolean,nil] true: success, false: failure, nil: failure+dont retry
+      def curl_download(url, filename, max_time: 300)
+        return nil unless self.class.enabled?
+        cmd = Yast::Builtins.sformat(
+          "/usr/bin/curl --location --verbose --fail --max-time %6 " \
+          "--connect-timeout 15  %1 '%2' --output '%3' > '%4/%5' 2>&1",
+          curl_proxy_args,
+          url,
+          Yast::String.Quote(filename),
+          Yast::String.Quote(Yast::Directory.logdir),
+          "curl_log",
+          max_time
+        )
+        ret = Yast::SCR.Execute(Yast::Path.new(".target.bash"), cmd)
+        log.info("#{cmd} returned #{ret}")
```

Comment:
NP: I'd remove `curl_proxy_args` from the logged string (`cmd.sub(curl_proxy_args, "")`), logging proxy passwords should be avoided.  (The `y2log` might be attached to bugzilla and the private data could leak.)

Reply:
Or replace with "@PROXYPASSWORD@" or something like that.